### PR TITLE
Add Github action for test coverage

### PR DESCRIPTION
**Pull request by vzhukovs**
### What does this PR do?
Provides an ability to generate test coverage report and upload it to https://codecov.io/gh

Signed-off-by: Vladyslav Zhukovskyi <vzhukovs@redhat.com>

### Screenshot/screencast of this PR

Add Github action for test coverage by vzhukovs · Pull Request #1000 · eclipse:che-theia 2021-02-23 00-04-41

### What issues does this PR fix or reference?
fixes https://github.com/eclipse/che/issues/18844

### How to test this PR?
There should be two new Github Actions that perform test coverage report generation.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -30,12 +30,16 @@ jobs:
...
+      with:
+        fetch-depth: "0"
     - uses: actions/setup-node@v1
       name: Configuring nodejs 10.x version
       with:
         node-version: '12.x'
     - name: build
       run: yarn
+    - name: Test Coverage
+      uses: codecov/codecov-action@v1
```
Comment (benoitf) [inline .github/workflows/pr.yml:42]: no need to specify location of coverage reports, it's finding it automatically ?

**Review by benoitf — APPROVED** [review]

File: .github/workflows/pr.yml (hunk 1 above)
Comment (vzhukovs) [reply .github/workflows/pr.yml:42]: From what I saw from logs, coverage reports are found automatically:

```
==> Searching for coverage reports in:
    + .
    -> Found 12 reports
==> Detecting git/mercurial file structure
==> Reading reports
    + ./generator/coverage/clover.xml bytes=40754
    + ./generator/coverage/coverage-final.json bytes=111291
    + ./generator/coverage/lcov.info bytes=15992
    + ./plugins/workspace-plugin/coverage/clover.xml bytes=32467
    + ./plugins/workspace-plugin/coverage/coverage-final.json bytes=102618
    + ./plugins/workspace-plugin/coverage/lcov.info bytes=14644
    + ./plugins/resource-monitor-plugin/coverage/clover.xml bytes=9776
    + ./plugins/resource-monitor-plugin/coverage/coverage-final.json bytes=27324
    + ./plugins/resource-monitor-plugin/coverage/lcov.info bytes=3888
    + ./plugins/ports-plugin/coverage/clover.xml bytes=26398
    + ./plugins/ports-plugin/coverage/coverage-final.json bytes=84172
    + ./plugins/ports-plugin/coverage/lcov.info bytes=12489
==> Appending adjustments
    https://docs.codecov.io/docs/fixing-reports
    + Found adjustments
==> Gzipping contents
        88K	/tmp/codecov.kOCN4l.gz
```

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -30,12 +30,16 @@ jobs:
     steps:
     - uses: actions/checkout@v2
       name: Checkout che-theia source code
+      with:
+        fetch-depth: "0"
```
Comment (azatsarynnyy) [inline .github/workflows/pr.yml:34]: why we need this input?

Comment (vzhukovs) [reply .github/workflows/pr.yml:34]: It is needed for correct fetching tags in current action: https://github.com/lerna/lerna/issues/2542

**Review by azatsarynnyy — APPROVED** [review]

**Pull request merged**